src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -257,68 +277,127 @@ export const useSwapper = () => {
-      try {
-        const swapper = await swapperManager.getBestSwapper({
-          buyAssetId: buyAsset.assetId,
-          sellAssetId: sellAsset.assetId,
-        })
+    debounce(
+      async ({
+        amount,
+        sellAsset,
+        buyAsset,
+        action,
+        wallet,
+        swapperManager,
+        accountSpecifiersList,
+        chainAdapterManager,
+      }: DebouncedQuoteInput) => {
+        try {
+          const swapper = await swapperManager.getBestSwapper({
+            buyAssetId: buyAsset.assetId,
+            sellAssetId: sellAsset.assetId,
+          })
 
-        if (!swapper) throw new Error('no swapper available')
+          if (!swapper) throw new Error('no swapper available')
+          const [sellAssetUsdRate, buyAssetUsdRate, feeAssetUsdRate] = await Promise.all([
+            swapper.getUsdRate({ ...sellAsset }),
+            swapper.getUsdRate({ ...buyAsset }),
+            swapper.getUsdRate({ ...feeAsset }),
+          ])
+
+          const { sellAmount, buyAmount, fiatSellAmount } = await calculateAmounts({
+            amount,
+            buyAsset,
+            sellAsset,
+            buyAssetUsdRate,
+            sellAssetUsdRate,
+            action,
+          })
 
-        const [sellAssetUsdRate, buyAssetUsdRate, feeAssetUsdRate] = await Promise.all([
-          swapper.getUsdRate({ ...sellAsset }),
-          swapper.getUsdRate({ ...buyAsset }),
-          swapper.getUsdRate({ ...feeAsset }),
-        ])
+          const { chainId: receiveAddressChainId } = fromAssetId(buyAsset.assetId)
 
-        const { sellAmount, buyAmount, fiatSellAmount } = await calculateAmounts({
-          amount,
-          buyAsset,
-          sellAsset,
-          buyAssetUsdRate,
-          sellAssetUsdRate,
-          action,
-        })
+          const chainAdapter = chainAdapterManager.get(receiveAddressChainId)
 
-        const tradeQuote: TradeQuote<KnownChainIds> = await (async () => {
-          if (sellAsset.chainId === KnownChainIds.EthereumMainnet) {
-            return swapper.getTradeQuote({
-              chainId: KnownChainIds.EthereumMainnet,
-              sellAsset,
-              buyAsset,
-              sellAmount,
-              sendMax: false,
-              sellAssetAccountNumber: 0,
-              wallet,
-            })
-          } else if (sellAsset.chainId === btcChainId) {
-            // TODO do bitcoin specific trade quote including `bip44Params`, `accountType` and `wallet`
-            // They will need to have selected an accountType from a modal if bitcoin
-            throw new Error('bitcoin unsupported')
-          }
-          throw new Error(`unsupported chain id ${sellAsset.chainId}`)
-        })()
+          if (!chainAdapter)
+            throw new Error(`couldnt get chain adapter for ${receiveAddressChainId}`)
 
-        await setFormFees({ trade: tradeQuote, sellAsset, tradeFeeSource: swapper.name })
+          // Get first specifier for receive asset chain id
```

Comment:
this is going to be an issue for everything except native wallet - `accountSpecifiers` only returns accounts for chains the wallet currently supports.

other than native and portis everything else needs to switch the chain on the wallet to repopulate account specifiers

outside the scope of this PR, but @shapeshift/product we will need some UX to pick a receive address, if the buy asset chain differs from the sell asset chain